packages/firebase_storage/lib/src/storage_metadata.dart

```diff
@@ -81,4 +83,7 @@ class StorageMetadata {
 
   /// The content type (MIME type) of the [StorageReference].
   final String contentType;
+
+  /// Custom metadata for the [StorageReference].
+  final Map<String, String> customMetadata;
```

Comment:
The field is `final`, but the map object is mutable. If it is not sensible to do
```dart
myStorageMetadata.customMetadata['hello'] = 'world';
```
we should encapsulate the map.